src/core/geometry/qgsgeos.cpp

```diff
@@ -58,10 +58,28 @@ static void throwGEOSException( const char *fmt, ... )
   va_end( ap );
 
   qWarning( "GEOS exception: %s", buffer );
+  QString message = QString::fromUtf8( buffer );
 
-  throw GEOSException( QString::fromUtf8( buffer ) );
+#ifdef _MSC_VER
+  // stupid stupid MSVC, *SOMETIMES* raises it's own exception if we throw GEOSException, resulting in a crash!
+  // see https://issues.qgis.org/issues/14752
+  // if you want to test alternative fixes for this, run the testqgsexpression.cpp test suite - that will crash
+  // and burn on the "line_interpolate_point point" test if a GEOSException is thrown.
+  try
+  {
+    throw GEOSException( message );
+  }
+  catch (...)
+  {
+    // oops, msvc threw an exception when we tried to throw the exception!
```

Comment:
Hi @nyalldawson I'm always impressed how you can find these fixes, congratulations... btw would make sense to add a TODO label in the code parts that have need more love. More love means: find the best fit to manage strange behaviour of MSVC.

Reply:
I wouldn't call this "fix" impressive by any means, but thanks anyway! I'll add the to-do :)

Reply:
I'm impressed how to have a debug session to find this kinds of errors :)